app/packages/state/src/recoil/selectors.ts

```diff
@@ -472,14 +472,27 @@ export const similarityMethods = selector<{
 export const extendedStagesUnsorted = selector({
   key: "extendedStagesUnsorted",
   get: ({ get }) => {
-    const sampleIds = get(atoms.extendedSelection)?.selection;
+    const extendedSelection = get(atoms.extendedSelection);
+    const sampleIds = extendedSelection?.selection;
+    const spatialSelection = extendedSelection?.spatialSelection;
     const extendedSelectionOverrideStage = get(
       atoms.extendedSelectionOverrideStage
     );
 
     if (extendedSelectionOverrideStage) {
       return extendedSelectionOverrideStage;
     }
+
+    if (spatialSelection) {
+      return {
+        "fiftyone.core.stages.GeoWithin": {
+          boundary: spatialSelection.polygon,
+          location_field: spatialSelection.field,
+          strict: true,
+          create_index: true,
```

Comment:
We should use `create_index=False` here and instead require users to create their own spatial index offline if they require it.

Note that I'm going to update this to `create_index=False` as well in a separate PR for the same reason:
https://github.com/voxel51/fiftyone/blob/5b055b70fa1cb5c5a25a0e5b8ea721f20ca4ee20/fiftyone/core/stages.py#L3370-L3371

Creating view stages shouldn't have expensive side effects like creating indexes.